src/userSync.js

```diff
@@ -0,0 +1,102 @@
+import * as utils from 'src/utils';
+
+const userSync = exports;
+// A queue of user syncs for each adapter
+const queue = {
+  image: []
+};
+
+/**
+ * @function fireSyncs
+ * @summary Trigger all user syncs in the queue
+ * @private
+ */
+function fireSyncs() {
+  try {
+    // Fire image pixels
+    queue.image.forEach((sync) => {
+      let bidderName = sync[0];
+      let trackingPixelUrl = sync[1];
+      utils.logMessage(`Invoking image pixel user sync for bidder: ${bidderName}`);
+      // insertAdjacentHTML expects HTML string - convert DOM object to string
+      let img = userSync.createImgObject(trackingPixelUrl);
+      if (img) {
+        utils.insertElement(img);
+      }
+    });
+    // Reset the image pixel queue
+    queue.image = [];
+  }
+  catch (e) {
+    utils.logError('Error firing user syncs', e);
+  }
+}
+
+/**
+ * @function hideAndIdElem
+ * @summary Modifies a DOM element to be hidden from user sight and adds a unique ID
+ * @private
+ * @params {object} elementNode A valid DOM element
+ * @returns {object} A valid DOM element
+ */
+function hideAndIdElem(elementNode) {
+  elementNode = elementNode.cloneNode();
+  elementNode.height = 0;
+  elementNode.width = 0;
+  elementNode.style.display = 'none';
+  elementNode.id = utils.getUniqueIdentifierStr();
+  return elementNode;
+}
+
+/**
+ * @function createImgObject
+ * @summary Create an img DOM element for sending a pixel. Made public for test purposes
+ * @public
+ * @params {string} url The URL for the image pixel
+ * @returns {object} A valid DOM element
+ */
+userSync.createImgObject = function(url) {
+  if (!url) {
+    return;
+  }
+  let img = hideAndIdElem(new Image());
+  img.src = encodeURI(url);
+  img.onload = function() {
+    // Once the sync is done remove the element
+    try {
+      let thisImg = document.getElementById(this.id);
+      thisImg.parentNode.removeChild(thisImg);
+    }
+    catch (e) {}
+  };
+  return img;
+};
+
+/**
+ * @function registerSync
+ * @summary Add sync for this bidder to a queue to be fired later
+ * @public
+ * @params {string} type The type of the sync including image, iframe, and ajax
+ * @params {string} bidder The name of the adapter. e.g. "rubicon"
+ * @params {string|object} data A series of arguments
+
+ * @example <caption>Using Image Sync</caption>
+ * // registerSync(type, adapter, pixelUrl)
+ * userSync.registerSync('image', 'rubicon', 'http://example.com/pixel')
+ */
+userSync.registerSync = function(type, bidder, ...data) {
+  if (!queue[type]) {
+    return utils.logWarn(`User sync type "{$type}" not supported`);
+  }
+  queue[type].push([bidder, ...data]);
+};
+
+/**
+ * @function syncUsers
+ * @summary Trigger all the user syncs based on publisher-defined timeout
+ * @public
+ * @params {int} timeout The delay in ms before syncing data - default 0
+ */
+userSync.syncUsers = function(timeout = 0) {
+  setTimeout(fireSyncs, timeout);
+};
```

Comment:
I think you hit the nail on the head. If _their average page load time_ is 5s, then a timeout of 5100s is a reasonable choice.

The timeout is just an indirect means of accomplishing the _real_ goal. In your example, it's "after page loads"

I'm just having a hard time imagining when it's the appropriate choice.

For simple pages, they should userSync on the `load` event.
For pages running scripts with a `setTimeout` call, they should userSync at the end of the callback. 
For complex pages which lots of js which needs to be executed in small bits to avoid page jank, then they should be adding the userSync calls to whatever async-queue they're using.

There may be pages out there where a fixed timeout is actually the best implementation choice... but I feel like I must raise the question: is it really a wise decision to hardcode that choice into prebid?